src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+            'iotCredEndpoint'] = iot_cred_endpoint
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'iotDataEndpoint'] = iot_data_endpoint
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'runWithDefault']['posixUser'] = group + ':' + user
+        data['services']['aws.greengrass.NucleusLite']['configuration'][
+            'iotRoleAlias'] = "ggl-uat-role-alias"
+        print(data)
+
+        with open(file_path, 'w') as file:
+            yaml.dump(data, file, default_flow_style=False)
+
+        print(f"Successfully modified the config file")
+        return True
+
+    except Exception as e:
+        print(f"Error when modifying config.yaml: {str(e)}")
+        return False
+
+
+def _stop_and_disable_services():
+    commands = [
+        ['sudo', 'systemctl', 'stop', 'greengrass-lite.target'],
+        ['sudo', 'systemctl', 'disable', 'greengrass-lite.target'],
+        ['sudo', 'systemctl', 'disable', 'ggl.aws_iot_tes.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.aws_iot_mqtt.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_config.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_health.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_fleet_status.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_deployment.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg_pubsub.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.gg-ipc.socket.socket'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggconfigd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.iotcored.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.tesd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggdeploymentd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.gg-fleet-statusd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggpubsubd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.gghealthd.service'],
+        ['sudo', 'systemctl', 'disable', 'ggl.core.ggipcd.service'],
+        ['sudo', 'systemctl', 'daemon-reload'],
+        ['sudo', 'rm', '-rf', '/etc/systemd/system/ggl*']
+    ]
+
+    for cmd in commands:
+        try:
+            subprocess.run(cmd, check=True)
+        except Exception as e:
+            print(f"Error when executing {' '.join(cmd)}: {e}")
+            return False
+    print("Successfully stopped and disabled all sevices")
+    return True
+
+
+# ===============================================
+# FILE & DIR FUNCTIONS
+# ===============================================
+
+
+def _create_dir(dir: str,
+                permission=False,
+                octal=None,
+                ownership=False,
+                flag=None,
+                user=None,
+                group=None) -> bool:
+    try:
+        if not os.path.exists(dir):
+            subprocess.run(['sudo', 'mkdir', dir], check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Starting a process with a partial executable path https://bandit.readthedocs.io/en/latest/plugins/b607_start_process_with_partial_path.html